motoko/http_counter/src/main.mo

```diff
@@ -0,0 +1,150 @@
+import Nat "mo:base/Nat";
+import Text "mo:base/Text";
+import Array "mo:base/Array";
+import Option "mo:base/Option";
+import Prim "mo:prim";
+import Prelude "mo:base/Prelude";
+
+
+actor HttpCounter {
+
+  type StreamingCallbackHttpResponse = {
+    body: Blob;
+    token: ?Token;
+  };
+
+  type Token = {
+    // Add whatever fields you'd like
+    arbitrary_data: Text;
+  };
+
+  type CallbackStrategy = {
+    callback: shared query (Token) -> async StreamingCallbackHttpResponse;
+    token: Token;
+  };
+
+  type StreamingStrategy =  {
+    #Callback: CallbackStrategy;
+  };
+
+  type HeaderField = (Text, Text);
+
+  type HttpResponse = {
+    status_code: Nat16;
+    headers: [HeaderField];
+    body: Blob;
+    streaming_strategy: ?StreamingStrategy;
+    upgrade: ?Bool;
+  };
+
+  type HttpRequest = {
+    method: Text;
+    url: Text;
+    headers: [HeaderField];
+    body: Blob;
+  };
+
+  func isGzip(x : HeaderField) : Bool {
+    Text.map(x.0 , Prim.charToLower) == "accept-encoding" and Text.contains(Text.map(x.1 , Prim.charToLower), #text "gzip");
```

Comment:
It's a shame you need to access Prim for this, but that's our fault and we should provide proper Char.toLower and toUpper implementations...